src/pytest/test.py

```diff
@@ -1002,6 +1066,53 @@ def testAddHash(env):
     env.assertEqual(1, res[0])
     env.assertEqual("doc2", res[1])
 
+def testSafeAddHash(env):
+    if env.is_cluster():
```

Comment:
Is this a copy/paste from addhash? if so, refactor into a single function.

Reply:
For the most part, it isn't.
It also has many errors catching tests.
My thinking was it would be better for future debugging.